tools/docker-compose-minikube/minikube/tasks/main.yml

```diff
@@ -0,0 +1,86 @@
+---
+- name: Set home binary path directory
+  set_fact:
+    home_bin: "{{ lookup('env','HOME') }}/bin"
+
+- name: Create home binary path directory
+  file:
+    path: "{{ home_bin }}"
+    state: 'directory'
+    mode: '0755'
+
+# Linux block
+- block:
+    - name: Download Minikube
+      get_url:
+        url: "{{ minikube_url_linux }}"
+        dest: "{{ home_bin }}/minikube"
+        mode: 0755
+
+    - name: Download Kubectl
+      get_url:
+        url: "{{ kubectl_url_linux }}"
+        dest: "{{ home_bin }}/kubectl"
+        mode: 0755
+  when:
+    - ansible_architecture == "x86_64"
+    - ansible_system == "Linux"
+
+# MacOS block
+- block:
+    - name: Download Minikube
+      get_url:
+        url: "{{ minikube_url_macos }}"
+        dest: "{{ home_bin }}/minikube"
+        mode: 0755
+
+    - name: Download Kubectl
+      get_url:
+        url: "{{ kubectl_url_macos }}"
+        dest: "{{ home_bin }}/kubectl"
+        mode: 0755
+  when:
+    - ansible_architecture == "x86_64"
+    - ansible_system == "Darwin"
+
+- name: Starting Minikube
+  shell: "{{ home_bin }}/minikube start --driver=docker --install-addons=true --addons=ingress"
+
+- name: Create ServiceAccount and clusterRoleBinding
+  k8s:
+    apply: yes
+    definition: "{{ lookup('template', 'rbac.yml.j2') }}"
+
+- name: Retrieve serviceAccount secret name
+  k8s_info:
+    kind: ServiceAccount
+    namespace: '{{ minikube_service_account_namespace }}'
+    name: '{{ minikube_service_account_name }}'
+  register: service_account
+
+- name: Register serviceAccount secret name
+  set_fact:
+    _service_account_secret_name: '{{ service_account["resources"][0]["secrets"][0]["name"] }}'
+  when:
+    - service_account["resources"][0]["secrets"] | length
+    - '"name" in service_account["resources"][0]["secrets"][0]'
+
+- name: Retrieve bearer_token from serviceAccount secret
+  k8s_info:
+    kind: Secret
+    namespace: '{{ minikube_service_account_namespace }}'
+    name: '{{ _service_account_secret_name }}'
+  register: _service_account_secret
+
+- name: Load Minikube CA certificate and bearer token
+  set_fact:
+    minikube_ca_cert: '{{ _service_account_secret["resources"][0]["data"]["ca.crt"] | b64decode }}'
+    service_account_token: '{{ _service_account_secret["resources"][0]["data"]["token"] | b64decode }}'
+  when:
+    - _service_account_secret["resources"][0]["data"] | length
+
+- name: Render minikube credential JSON template
+  template:
+    src: api_credential.json.j2
+    dest: "api_credential.json"
+    mode: '0600'
```

Comment:
maybe all this regarding the credential can be carved out into a separate playbook and then you use the awx.awx collection to create the credential, and then have the playbook run after the awx API is up or just leave it as a tool someone can run given they have minikube and awx up and running. Using the awx.awx.credential module would implicate deleting tools/docker-compose-minikube/minikube/templates/api_credential.json.j2 and the service definition you have in tools/docker-compose/supervisor.conf 
https://docs.ansible.com/ansible/latest/collections/awx/awx/credential_module.html#ansible-collections-awx-awx-credential-module

Reply:
Having a separate tool to be executed once the AWX API is up will facilitate a lot of things. If we are good with that extra step, I can adjust this into a separate play.

Reply:
So to avoid an extra step, decided to call it a directory on the `boostrap_development.py` script which works transparently.